plansys2_bt_actions/include/plansys2_bt_actions/BTActionNode.hpp

```diff
@@ -82,7 +78,11 @@ class BtActionNode : public BT::ActionNodeBase
   {
     BT::PortsList basic = {
       BT::InputPort<std::string>("server_name", "Action server name"),
-      BT::InputPort<std::chrono::milliseconds>("server_timeout")
+      BT::InputPort<unsigned int>(
```

Comment:
There isn't an automatic conversion to `std::chrono::milliseconds` from a blackboard value (by my understanding), so a template specialization for the BehaviorTree.CPP function `convertFromString` would have to be implemented to specify a value in the xml, see [here](https://www.behaviortree.dev/tutorial_03_generic_ports/) for more details.

I see no reason to do this since we can just switch the type, but that is a point of opinion.